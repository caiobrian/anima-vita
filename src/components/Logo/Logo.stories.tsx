import { Logo } from '.'
import { Meta, Story } from '@storybook/react'
import { LogoProps } from './Logo'

export default {
  title: 'Logo',
  component: Logo,
  argTypes: {
    size: {
      control: { type: 'select', options: ['small', 'medium', 'large'] }
    }
  }
} as Meta

const Template: Story<LogoProps> = (args) => <Logo {...args} />

export const Default = Template.bind({})

Default.args = {
  size: 'medium'
}
