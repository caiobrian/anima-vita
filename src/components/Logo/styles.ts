import styled, { css } from 'styled-components'
import { LogoProps } from './Logo'

const handleSize = {
  small: css`
    width: 8rem;
  `,
  medium: css`
    width: 16rem;
  `,
  large: css`
    width: 32rem;
  `
}

const Image = styled.img<LogoProps>`
  ${({ size }) => handleSize[size]}
`

export { Image }
