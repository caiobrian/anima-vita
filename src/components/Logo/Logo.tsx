import { AnimaLogo } from '../../assets/svg'
import { Image } from './styles'

export type LogoProps = {
  size: 'small' | 'medium' | 'large'
}

export const Logo = ({ size = 'medium' }: LogoProps) => (
  <Image src={AnimaLogo} alt="logo" size={size} />
)
