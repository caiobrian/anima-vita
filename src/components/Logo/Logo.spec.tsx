import { render } from '@testing-library/react'
import { Logo } from './Logo'

describe('Logo', () => {
  it('should render correctly', () => {
    const { container } = render(<Logo size="medium" />)

    expect(container).toMatchSnapshot()
  })
})
