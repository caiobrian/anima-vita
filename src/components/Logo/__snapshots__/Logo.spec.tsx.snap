// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should render correctly 1`] = `
<div>
  <img
    alt="logo"
    class="sc-bdvvtL emhHcc"
    src="default"
  />
</div>
`;
